--- a/src/main/java/com/minerl/multiagent/env/EnvServer.java
+++ b/src/main/java/com/minerl/multiagent/env/EnvServer.java
@@ -768,8 +768,8 @@
         boolean withInfo = options == 0 || options == 2;
         envTickCounter = PlayRecorder.getInstance().getTickCounter();
         execActions(actions, options);
-        awaitPendingFlintAndSteelMarshal();
         waitForNextObservation();
+        awaitPendingFlintAndSteelMarshal();
         byte[] obs = getPOVObservation();
         boolean done = mc.player == null || !mc.player.isAlive() || (this.doneOnDeath && mc.isHasPlayerRespawned());
